### PR TITLE
[AMLII-2036] Add check for readonly file systems

### PR DESCRIPTION
**Pull request by soberpeach**
### What does this PR do?

Returns a valid launcher even if agent is in a readonly file system. Errors are handled gracefully instead of causing a panic.

### Motivation

Bug discovered when then agent was run in a readonly filesystem.

### Describe how to test/QA your changes

Run the agent with logs enabled and make the agent logs run directory readonly (directory is located at `/opt/datadog-agent-/run` on mac) and ensure the agent doesn't panic on startup.

### CODE REVIEW

**Review by rayz — APPROVED** [review]

File: releasenotes/notes/fix-panic-on-readonly-filesystems-1012ba761aa70aaa.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fixes a panic caused by running the agent on readonly filesystems. The
```
Comment (jhgilbert) [inline releasenotes/notes/fix-panic-on-readonly-filesystems-1012ba761aa70aaa.yaml:11]:
```suggestion
    Fixes a panic caused by running the Agent on readonly filesystems. The
```

File: releasenotes/notes/fix-panic-on-readonly-filesystems-1012ba761aa70aaa.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    agent will now return integration launchers and handle memory gracefully.
```
Comment (jhgilbert) [inline releasenotes/notes/fix-panic-on-readonly-filesystems-1012ba761aa70aaa.yaml:12]:
```suggestion
    Agent will now return integration launchers and handle memory gracefully.
```

**Review by jhgilbert — APPROVED** [review]
Approved with minor suggestions, thanks!

**Pull request merged**